### PR TITLE
LRAC-7934 Add a DXP data source and assert the source counter for the attributes in Settings > Definitions > Individuals is incremented

### PR DESCRIPTION
**Pull request by r4ddek**
**New test cases:**
IndividualsDefinitions.testcase >
test AssertNewSourceInSourceListForIndividualAttribute

**Macros and tests refactored:**
None.

**New macros:**
ACDefinitions.macro >
assertIndividualAttributesSource();
goToDefinitionType();
openIndividualAttributeDetails();

ACUtils.macro >
clearSearchBar();
closeModal();

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IndividualDefinitions.testcase [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		// Open Analytics Cloud, login, open faro demo workspace and take the token to connect to DXP
+
+		var analyticsCloudURL = PropsUtil.get("analytics.cloud.url");
+		Navigator.openURL(baseURL = "${analyticsCloudURL}");
+
+		// Take token from Analytics Cloud
+
+		ACLogin.loginAs();
+
+		ACWorkspace.openWorkspace();
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IndividualDefinitions.testcase]: Hi @r4ddek , these calls can be replaced by `ACUtils.launchAC()`;

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IndividualDefinitions.testcase [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+		ACSettings.takeTokenFromAnalyticsCloud();
+
+		// Go to DXP portal
+
+		TestCase.setUpPortalInstance();
+
+		Navigator.openURL();
+
+		User.firstLoginPG();
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IndividualDefinitions.testcase]: Hi @r4ddek , these two calls can be replaced by `ACUtils.launchDXP()`;

File: portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/ACUtils.path [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+<!-- CLOSE MODAL -->
+<tr>
+    <td>MODAL_CLOSE_BUTTON</td>
+    <td>//button[contains(@class, 'close')]</td>
+    <td></td>
+</tr>
+<!-- SEARCH_BAR -->
+<tr>
+    <td>CLEAR_SEARCH_BAR_BUTTON</td>
+    <td>//div[contains(@class, 'input-group-item')]//button[contains(@class, 'button-root btn btn-unstyled')]</td>
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/ACUtils.path]: hi @r4ddek , when using contains, try to not use multiple classes/elements to not let your selector fragile. In this case, you can just have in the second part of your selector:
`//button[contains(@class, 'button-root')]`

**Review by alexcandidos — COMMENTED** [review]
Hi @r4ddek , the code looks good to me, I just left some comments/suggestions to improve the code readability. Thanks!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IndividualDefinitions.testcase (hunk 1 above)
Comment (r4ddek) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IndividualDefinitions.testcase]: Thanks for the update, Alex! :)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IndividualDefinitions.testcase (hunk 2 above)
Comment (r4ddek) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/IndividualDefinitions.testcase]: Changed, thanks :)

File: portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/ACUtils.path (hunk 3 above)
Comment (r4ddek) [reply portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/ACUtils.path]: Ops, this was a leftover from a revert, thanks!!

**Comment by r4ddek** [comment]
ci:forward

**Pull request closed without merging**